pkg/ebpf/c/tracee.bpf.c

```diff
@@ -963,14 +1000,61 @@ BPF_HASH(module_init_map, u32, kmod_data_t, 256);                  // holds modu
 BPF_LRU_HASH(fd_arg_path_map, fd_arg_task_t, fd_arg_path_t, 1024); // store fds paths by task
 BPF_LRU_HASH(bpf_attach_map, u32, bpf_attach_t, 1024);             // holds bpf prog info
 BPF_LRU_HASH(bpf_attach_tmp_map, u32, bpf_attach_t, 1024);         // temporarily hold bpf_attach_t
+BPF_HASH(errors_count, bpf_error_t, bpf_error_count_t, 4096);      // errors count
+BPF_PERCPU_ARRAY(error_output_scratch, bpf_error_output_t, 1);     // error output scratch
 // clang-format on
 
 // EBPF PERF BUFFERS -------------------------------------------------------------------------------
 
+BPF_PERF_OUTPUT(errors, 1024);      // errors submission
 BPF_PERF_OUTPUT(events, 1024);      // events submission
 BPF_PERF_OUTPUT(file_writes, 1024); // file writes events submission
 BPF_PERF_OUTPUT(net_events, 1024);  // network events submission
 
+// HELPERS: ERRORS ---------------------------------------------------------------------------------
+
+static __always_inline void
+do_tracee_error(void *ctx, u32 opts, enum bpf_error_id id, s64 ret, u32 line, void *file)
+{
+    if (!ctx || !file)
+        return;
+
+    u32 zero = 0;
+    bpf_error_output_t *err_output = bpf_map_lookup_elem(&error_output_scratch, &zero);
+    if (unlikely(err_output == NULL))
+        return;
+
+    err_output->err.id = id;
+    err_output->err.ret = ret;
+    err_output->err.cpu = bpf_get_smp_processor_id();
+    err_output->err.line = line;
+    READ_KERN_STR_INTO(err_output->err.file, file);
+
+    bpf_error_count_t counter_buf = {};
+    counter_buf.count = 1;
+    counter_buf.ts = bpf_ktime_get_ns(); // store the current ts
+    u64 ts_prev = 0;
+
+    bpf_error_count_t *counter = bpf_map_lookup_elem(&errors_count, &err_output->err);
+    if (likely(counter != NULL)) {
+        ts_prev = counter->ts; // store previous ts
+
+        counter->count += 1;
+        counter->ts = counter_buf.ts; // set to current ts
```

Comment:
@yanivagman I overlooked this part. Probably it's racy. WDYT about reverting it to my previous `BPF_PERCPU_HASH(errors_count, bpf_error_t, bpf_error_count_t, 4096);`?

Reply:
Can you please explain?

Reply:
Ignore this. errors_count key has the cpu id so each cpu has its own error count per origin and return value.